apps/voting-disputable/test/helpers/deployer.js

```diff
@@ -75,21 +98,35 @@ class VotingDeployer {
     const receipt = await this.dao.newAppInstance(appId, this.base.address, '0x', false, { from: owner })
     const voting = await this.base.constructor.at(await getInstalledApp(receipt, appId))
 
-    const restrictedPermissions = ['SET_AGREEMENT_ROLE', 'MODIFY_SUPPORT_ROLE', 'MODIFY_QUORUM_ROLE', 'MODIFY_OVERRULE_WINDOW_ROLE', 'MODIFY_EXECUTION_DELAY_ROLE', 'MODIFY_QUIET_ENDING_CONFIGURATION']
+    const restrictedPermissions = ['MODIFY_SUPPORT_ROLE', 'MODIFY_QUORUM_ROLE', 'MODIFY_OVERRULE_WINDOW_ROLE', 'MODIFY_EXECUTION_DELAY_ROLE', 'MODIFY_QUIET_ENDING_CONFIGURATION']
     await this._createPermissions(voting, restrictedPermissions, owner)
 
     const openPermissions = ['CREATE_VOTES_ROLE', 'CHALLENGE_ROLE']
     await this._createPermissions(voting, openPermissions, ANY_ENTITY, owner)
 
+    if (!this.agreement) await this.deployAgreement(options)
+    await this._createPermissions(voting, ['SET_AGREEMENT_ROLE'], this.agreement.address, owner)
+
     if (currentTimestamp) await voting.mockSetTimestamp(currentTimestamp)
     this.previousDeploy = { ...this.previousDeploy, voting }
     return voting
   }
 
-  async deployBase(options = {}) {
-    const agreement = options.agreement || false
-    const contractName = `DisputableVoting${agreement ? 'Mock' : 'WithoutAgreementMock'}`
-    const Voting = this._getContract(contractName)
+  async deployAgreement(options = {}) {
+    const owner = options.owner || await this._getSender()
+    if (!this.dao) await this.deployDAO(owner)
+    if (!this.base) await this.deployBase(options)
+
+    const { currentTimestamp } = { ...DEFAULT_VOTING_INITIALIZATION_PARAMS, ...options }
+    agreementDeployer.previousDeploy = { ...agreementDeployer.previousDeploy, dao: this.dao, acl: this.acl, owner }
+    const agreement = await agreementDeployer.deployAndInitializeAgreementWrapper({ ...options, owner, currentTimestamp })
```

Comment:
I'm not sure about this—doesn't this move all of our tests to be integration tests, since we're using the real agreement?

We should definitely have a suite of these with a real agreement, but I was wondering if it would be easier for us to test using a mocked version for the unit tests?